Sources/SwiftProtobuf/DelimitedMessageUtil.swift

```diff
@@ -0,0 +1,215 @@
+  /// exactly that size.
+  ///
+  /// - Parameters:
+  ///   - messageType: The type of message to read.
+  ///   - from: The `InputStream` to read the data from.  The stream is assumed
+  ///     to be ready to read from.
+  ///   - extensions: An `ExtensionMap` used to look up and decode any
+  ///     extensions in this message or messages nested within this message's
+  ///     fields.
+  ///   - partial: If `false` (the default), this method will check
+  ///     `Message.isInitialized` before encoding to verify that all required
+  ///     fields are present. If any are missing, this method throws
+  ///     `BinaryEncodingError.missingRequiredFields`.
+  ///   - options: The BinaryDecodingOptions to use.
+  /// - Returns: The message read.
+  /// - Throws: `BinaryDecodingError` if decoding fails; throws
+  ///           `DelimitedMessageError` for some rading errors, and the
+  ///           underlying InputStream.streamError for a stream error.
+  public static func parse<M: Message>(
+    messageType: M.Type,
+    from stream: InputStream,
+    extensions: ExtensionMap? = nil,
+    partial: Bool = false,
+    options: BinaryDecodingOptions = BinaryDecodingOptions()
+  ) throws -> M {
+    var message = M()
+    try parse(mergingTo: &message,
+              from: stream,
+              extensions: extensions,
+              partial: partial,
+              options: options)
+    return message
+  }
+
+  /// Updates the message by reading a single size-delimited message from
+  /// the given stream. Delimited format allows a single file or stream to
+  /// contain multiple messages, whereas normally parsing consumes the entire
+  /// input. A delimited message is a varint encoding the message size
+  /// followed by a message of exactly that size.
+  ///
+  /// - Note: If this method throws an error, the message may still have been
+  ///   partially mutated by the binary data that was decoded before the error
+  ///   occurred.
+  ///
+  /// - Parameters:
+  ///   - mergingTo: The message to merge the data into.
+  ///   - from: The `InputStream` to read the data from.  The stream is assumed
+  ///     to be ready to read from.
+  ///   - extensions: An `ExtensionMap` used to look up and decode any
+  ///     extensions in this message or messages nested within this message's
+  ///     fields.
+  ///   - partial: If `false` (the default), this method will check
+  ///     `Message.isInitialized` before encoding to verify that all required
+  ///     fields are present. If any are missing, this method throws
+  ///     `BinaryEncodingError.missingRequiredFields`.
+  ///   - options: The BinaryDecodingOptions to use.
+  /// - Throws: `BinaryDecodingError` if decoding fails; throws
+  ///           `DelimitedMessageError` for some rading errors, and the
+  ///           underlying InputStream.streamError for a stream error.
+  public static func parse<M: Message>(
```

Comment:
Since the methods are on the scoping type, I was worried "merge" could be seen as an input or output operation.  But maybe not?